### DISCUSSION TITLE
Controlling Model Size in AR Environment

### DISCUSSION THREAD

**Discussion by abhirami100** [google/model-viewer#4709] [Q&A]
I'm encountering an issue with controlling the size of models. Specifically, I'm trying to display a chair in the AR view at a standard size, but no matter what adjustments I make to the model's size, it always appears smaller than I'd like in the AR environment.

I've tried scaling the model up, but when I view it in the AR viewer, it seems to default to its original size. Is there a way to set a fixed size for a model in AR so that it consistently appears at the desired scale?

**Comment by danilokrulj** [comment]
You set the correct scale while editing the model (in blender or some other 3D program). And you add ar-scale="fixed" to the <model-viewer> element, that will keep your model at 100% scale.

PS.
It might seem smaller/larger if tracking doesn't get the ground plane right, but if you scan enough it should correct it.

**Reply by danilokrulj** [reply]
Do you change the scale/size of the model in the code? If you change it through code the changes wouldn't be visible in SceneViewer, I think, SceneViewer shows the original model without any adjustments made through code.

**Reply by abhirami100** [reply]
After attempting to scale the model correctly in Blender and  set ar-scale="fixed", I encountered the same problem where the model remained  small.

**Reply by danilokrulj** [reply]
Sounds weird, do you think its a code problem or a problem with the model? Are you sure you are using correct measurements units?
Maybe try putting your model into this code and see if the dimensions are the same as in blender and as you expected.

https://modelviewer.dev/examples/annotations/index.html#dimensions

**Reply by elalish** [reply]
Yes, you can also see the dimensions by clicking "Info" after dropping your model in our editor: [https://modelviewer.dev/editor](https://modelviewer.dev/editor)

Also keep in mind that the model will only look the right size if it's actually sitting on the floor (which it's not to start with). You need to move your phone around to allow AR to find your floor. Hence the utility of a prompt: https://modelviewer.dev/examples/augmentedreality/index.html#webXR

**Reply by abhirami100** [reply]
When I verify that the dimensions match those in Blender, and they do, do I need to adjust any settings so that if I scan a closer area, the model will appear larger ?
For example: https://ar.herabathroom.com/bathtub.html

**Reply by elalish** [reply]
Can you paste in some screen recordings so we can tell what problem you're seeing? I bet you're just not finding the floor (notoriously finicky with AR).

**Reply by abhirami100** [reply]
https://github.com/google/model-viewer/assets/110659626/07a5ba7b-3daf-4f0b-bf56-6a2627087411


Here is the code,
<model-viewer
  src="https://cdn.glitch.global/298a568e-26d1-4b26-80ab-19ad8a8d013b/sofa_chair.glb?v=1710419019526"
  ar
  ar-scale="fixed"
  camera-controls
  touch-action="pan-y"
  alt="A 3D model of an chair"
  shadow-intensity="2"
></model-viewer>

**Reply by elalish** [reply]
Yes, it's exactly as I thought - ARCore hasn't found your room's floor yet, because it requires parallax to do so. You have to move your phone side-to-side (not just turn it) for this to work, and you'll know it happens when the model jumps up / down to the floor height and the shadow shows up. 

Take a look at how the AR prompt is designed in my above link - you may want to copy that or make something similar to help your users.

**Reply by abhirami100** [reply]
Even with proper scanning and positioning attempts, the issue persists with this code. However, when I use "ar-modes='scene-viewer" ,the model displays in the correct size.
